tox.ini

```diff
@@ -20,6 +20,139 @@ setenv =
   PEP517_OUT_DIR = {env:PEP517_OUT_DIR:{toxinidir}{/}dist}
 
 
+[testenv:build-docs]
+allowlist_externals =
+  git
+basepython = python3
+commands_pre =
+  # Paramiko:
+  {envpython} -m pip install -r{toxinidir}/requirements-tunnel.txt
+commands =
+  # Retrieve possibly missing commits:
+  -git fetch --unshallow
+  -git fetch --tags
+
+  # Build the html docs with Sphinx:
+  {envpython} -m sphinx \
+    -j auto \
+    -b html \
+    {tty:--color} \
+    -a \
+    -n \
+    -W --keep-going \
+    -d "{temp_dir}/.doctrees" \
+    . \
+    "{envdir}/docs_out"
+
+  # Print out the output docs dir and a way to serve html:
+  -{envpython} -c\
+  'import pathlib;\
+  docs_dir = pathlib.Path(r"{envdir}") / "docs_out";\
+  index_file = docs_dir / "index.html";\
+  print("\n" + "=" * 120 +\
+  f"\n\nDocumentation available under:\n\n\
+  \tfile://\{index_file\}\n\nTo serve docs, use\n\n\
+  \t$ python3 -m http.server --directory \
+  \N\{QUOTATION MARK\}\{docs_dir\}\N\{QUOTATION MARK\} 0\n\n" +\
+  "=" * 120)'
+changedir = {toxinidir}/docs
+deps =
+  -r{toxinidir}/docs/requirements.txt
+  # FIXME: re-enable the "-r" + "-c" paradigm once the pip bug is fixed.
+  # Ref: https://github.com/pypa/pip/issues/9243
+  # -r{toxinidir}/docs/requirements.in
+  # -c{toxinidir}/docs/requirements.txt
+description = Build The Docs
+isolated_build = true
+passenv =
+  SSH_AUTH_SOCK
+skip_install = false
+usedevelop = false
+
+
+[testenv:doctest-docs]
```

Comment:
@abhinavsingh with this, you can start adding doctests as part of the docs snippets per https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html. It's important to know that those snippets aren't out-of-date and still work.